docs/website/mkdocs.yml

```diff
@@ -181,6 +181,7 @@ nav:
           # Dialects from llvm-external-projects/iree-dialects/
           - "Public dialects":
               - IREEInput: "reference/mlir-dialects/IREEInput.md"
+      - "Compiler passes": "reference/Passes.md"
```

Comment:
Presumably we'll also want to show pass documentation for the other dialects and parts of the compiler too?

Do you want a single page with all passes, or some other grouping?

Ideas:

* (A) Put passes on the same pages as dialects, if there is a 1:1 mapping.
* (B) Put passes next to dialects, still under the `mlir-dialects/` category (https://iree.dev/reference/mlir-dialects/ would then have https://iree.dev/reference/mlir-dialects/HAL _and_ https://iree.dev/reference/mlir-dialects/HALPasses, for example)
* (C) Put passes into a new category - `mlir-passes`, so we'd have an index page at https://iree.dev/reference/mlir-passes/ and then child pages like https://iree.dev/reference/mlir-passes/HAL

I'm leaning towards option C, but it would help to have a list of which pages would be added and how many passes there are in each before deciding.

Reply:
I cargo culted form MLIR's documentation organization. There all passes are on one [page](https://mlir.llvm.org/docs/Passes/). I am not sure how it would be more convenient for someone reading the doc. I imagine if you are interested in one specific dialect you would want them on the dialect page. If you don't know where is the pass you are looking for, you would want a single page. We may have passes that are not dialect related. If we move passes to the respective dialect pages we will still end up with some passes that we must put somewhere else.
If we put them on the dialect page (A), then we would need to stich them after generation into one file. Or find a way to tell `mlir-tablegen` to not put a `[TOC]` on the dialect page and then have a page with TOC that includes the dialect and pass pages, so it produces complete/correct TOC.

(C) looks good. I can reorganize the PR. I imagine some categories will have a few and some many passes, but the total should not be more than a few hundred. After all someone needs to write them.

Reply:
> If you don't know where is the pass you are looking for, you would want a single page.

Perhaps, but unlike the upstream MLIR docs, we do also have site-wide search: 
![image](https://github.com/user-attachments/assets/684b9c8c-1e46-4a29-aa67-d6584138c784)

> We may have passes that are not dialect related. If we move passes to the respective dialect pages we will still end up with some passes that we must put somewhere else.

Good point. Maybe we could group by compilation phase? And "other" for general passes with no better fit. I think at least that's a good enough argument against putting the passes in or next to the dialects.

We could also start with all the passes on a single page and then split them out later.

I'd at least want to see the MLIR passes show up in the nav as `:simple/llvm: MLIR passes` here: 
![image](https://github.com/user-attachments/assets/3cf7b19b-a120-44f7-a2d5-fb6563af55cd)

`mkdocs.yml`:
```yml
      - "MLIR passes":
          - "reference/mlir-passes/index.md"
```

`reference/mlir-passes/index.md`:
```markdown
---
icon: simple/llvm
---

# MLIR passes
```

Reply:
(marking this "reviewed" so it drops from my https://github.com/pulls/review-requested queue until this is addressed)